MinStack.java

```diff
@@ -0,0 +1,51 @@
+// Time Complexity :O(n)

```

Comment:
This defeats the purpose of the question. you cannot search thru the stack for the minimum element
try to complete this in constant time